records/state/state.go

```diff
@@ -46,9 +46,22 @@ type Label struct {
 
 // Status holds a task status as defined in the /state.json Mesos HTTP endpoint.
 type Status struct {
-	Timestamp float64 `json:"timestamp"`
-	State     string  `json:"state"`
-	Labels    []Label `json:"labels,omitempty"`
+	Timestamp       float64         `json:"timestamp"`
+	State           string          `json:"state"`
+	Labels          []Label         `json:"labels,omitempty"`
+	ContainerStatus ContainerStatus `json:"container_status"`
```

Comment:
In this case there's no difference; either way of representing "not supplied" seems equally OK to me.  This is because proper defensive practice would dictate checks for the key as well as a check for null (it's a valid JSON value after all).  Either way, I don't have a strong opinion.  As mentioned above, this doesn't appear to be used for serialization.  Please advise one way to proceed.

> On Sep 17, 2015, at 01:36, Tomás Senart <notifications@github.com> wrote:
> 
> In records/state/state.go:
> 
> > @@ -46,9 +46,22 @@ type Label struct {
> >  
> >  // Status holds a task status as defined in the /state.json Mesos HTTP endpoint.
> >  type Status struct {
> > -	Timestamp float64 `json:"timestamp"`
> > -	State     string  `json:"state"`
> > -	Labels    []Label `json:"labels,omitempty"`
> > +	Timestamp       float64         `json:"timestamp"`
> > +	State           string          `json:"state"`
> > +	Labels          []Label         `json:"labels,omitempty"`
> > +	ContainerStatus ContainerStatus `json:"container_status"`
> I personally don't like to use omitempty because when a field is absent in a JSON payload, one can't tell the difference between the field not existing and being empty for a particular instance.
> 
> —
> Reply to this email directly or view it on GitHub.
>